tests/sentry/models/test_projectcounter.py

```diff
@@ -14,3 +20,43 @@ def test_increment(default_project, upsert_sample_rate):
 
     assert Counter.increment(default_project, 42) == 42
     assert Counter.increment(default_project, 1) == 43
+
+
+@contextmanager
+def patch_group_creation(monkeypatch):
+    group_creation_results: list[Group | Exception] = []
+    group_creation_spy = MagicMock(
+        side_effect=capture_results(Group.objects.create, group_creation_results)
+    )
+    monkeypatch.setattr("sentry.event_manager.Group.objects.create", group_creation_spy)
+
+    yield (group_creation_spy, group_creation_results)
+
+
+def create_existing_group(project, monkeypatch, message):
+    with patch_group_creation(monkeypatch) as patches:
+        group_creation_spy, group_creation_results = patches
+
+        event = save_new_event({"message": message}, project)
+        group = Group.objects.filter(id=event.group_id).first()
```

Comment:
Nit: Should we just use `.get()` here?

Reply:
I thought that migration we worked on failed because we were using `.get()`. Am I remembering wrong?

Reply:
You're remembering right, but the context around that was that we had hardcoded IDs that didn't exist locally, so the exception being thrown caused the failure.

If the group doesn't exist here, we'll also have a bug because we're not checking for None. .get() is probably most correct here

Reply:
Ah, I see. All right, `.get()` it is!